server/xpub-model/entities/manuscript/index.js

```diff
@@ -16,67 +14,145 @@ const mergeObjects = (...inputs) =>
+        createdBy: { type: 'uuid' },
+        journalId: { type: 'uuid' },
+        status: { type: 'string' },
+        meta: {
+          title: { type: 'string' },
+          articleType: { type: 'string' },
+          articleIds: { type: 'array' },
+          abstract: { type: 'string' },
+          subjects: { type: 'array' },
+          notes: { type: 'array' },
+        },
+        previouslyDiscussed: { type: ['string', 'null'] },
+        previouslySubmitted: { type: 'array' },
+        cosubmission: { type: 'array' },
+        suggestionsConflict: { type: 'boolean' },
+        coverLetter: { type: 'string' },
+        opposedSeniorEditorsReason: { type: 'string' },
+        opposedReviewingEditorsReason: { type: 'string' },
+        opposedReviewersReason: { type: 'string' },
+        submitterSignature: { type: 'string' },
+        disclosureConsent: { type: 'boolean' },
+        formState: { type: ['string', 'null'] },
+
+        relatedManuscripts: { type: ['array', 'null'] },
+        qcIssues: { type: ['array', 'null'] },
+        decision: { type: ['string', 'null'] },
+        previousVersion: { type: 'uuid' },
+      },
+    }
+  }
+
+  static get jsonAttributes() {
+    // tell objection to serialise arrays as a postgres array rather than JSON
+    return []
+  }
+
+  static get relationMappings() {
+    return {
+      files: {
+        relation: BaseModel.HasManyRelation,
+        modelClass: `${__dirname}/../file`,
+        join: {
+          from: 'manuscript.id',
+          to: 'file.manuscriptId',
+        },
+      },
+      teams: {
+        relation: BaseModel.HasManyRelation,
+        modelClass: `${__dirname}/../team`,
+        join: {
+          from: 'manuscript.id',
+          to: 'team.objectId',
+        },
+      },
     }
+  }
 
-    if (manuscript.teams) {
-      await Promise.all(
-        manuscript.teams.map(team =>
-          TeamManager.save({ ...team, objectId: id, objectType: 'manuscript' }),
-        ),
-      )
+  static get statuses() {
+    return {
+      INITIAL: 'INITIAL',
+      MECA_EXPORT_PENDING: 'MECA_EXPORT_PENDING',
+      MECA_EXPORT_FAILED: 'MECA_EXPORT_FAILED',
+      MECA_EXPORT_SUCCEEDED: 'MECA_EXPORT_SUCCEEDED',
+      MECA_IMPORT_FAILED: 'MECA_IMPORT_FAILED',
+      MECA_IMPORT_SUCCEEDED: 'MECA_IMPORT_SUCCEEDED',
     }
+  }
+
+  static get MAX_SUGGESTED_REVIEWERS() {
+    return 6
+  }
+  static get MIN_SUGGESTED_REVIEWERS() {
+    return 3
+  }
 
-    if (manuscript.files) {
-      await Promise.all(
-        manuscript.files.map(file =>
-          FileManager.save({ ...file, manuscriptId: id }),
-        ),
-      )
+  static async find(id, user) {
+    const [manuscript] = await this.query().where({
+      'manuscript.id': id,
+      'manuscript.created_by': user,
+    })
+
+    if (!manuscript) {
+      throw new Error(`${this.name} not found`)
     }
+    // todo why does eager loading sometimes not work?
+    await manuscript.$loadRelated('[teams, files]')
```

Comment:
What is meant by eager loading? Do we expect relations to be resolved without explicitly doing it ourselves?

Reply:
I've managed to reproduce the issue locally.